keps/sig-windows/windows-runtimeclass-support.md

```diff
@@ -0,0 +1,575 @@
+
+## Motivation
+
+There are a few related customer experience problems when running mixed (multiple-OS and/or multiple-CPU-arch) clusters today. Kubernetes scheduling is based solely on what's declared in the deployment as part of the Pod spec. A Pod could be scheduled to an incompatible node and fail to pull. This error causes a retry and backoff loop, and doesn't automatically fail over to another node.
+
+In addition to the stuck pull failure case, there's also a question of user intent. Some containers are published for multiple OS's and/or architectures - this often referred to as multi-arch but the OCI distribution spec actually includes the OS, OS version, CPU architecture and variant. If a node can run multiple permutations of these, then there's not a deterministic way to know what the user intended to run. Running a container under CPU emulation may have performance penalties, or cost more in terms of leased compute time.
+
+Today with only a single supported Windows version (10.0.17763) & runtime (Docker), the problem is easily mitigated with NodeSelectors or Taints. This is documented today in [Guide for scheduling Windows containers in Kubernetes]. Moving these to RuntimeClasses simplifies the experience further, and can simplify deployment YAML for users in Kubernetes v1.16.
+
+Next, SIG-Windows is adding support for [Windows CRI-ContainerD], Windows nodes will be able to handle running multiple Windows OS versions using Hyper-V isolation. This technology could be used to run Linux containers as well, leading to more ambiguity. This KEP aims to resolve the ambiguity of pulling and running a container image and pod when multiple os/versions/architecture/variants may be supported on a single node.
+
+### Goals
+
+- Schedule a Windows Pod to a compatible node
+  - Allow matching Windows versions without Hyper-V isolation
+  - Allow opt-in on a per-Pod basis to run containers using existing Windows versions with backwards-compatibility provided by Hyper-V on a new Windows OS version node
+- Be able to schedule a specific image from a multi-arch manifest on a given node
+- Provide a simpler experience (fewer lines of YAML) than adding os and version nodeSelector and tolerations to each Pod
+
+### Non-Goals
+
+- Linux container support on Windows is not a requirement or test target for Kubernetes 1.17, but it's not specifically excluded.
+- Running newer Windows OS version containers (Windows Server version 1903) on an older OS version host (Windows Server version 1809). This is not supported today with or without Hyper-V isolation (see [Windows container version compatibility]).
+
+## Proposal
+
+For Kubernetes 1.17, we're proposing three key things to improve the experience deploying Windows containers across multiple versions, and enable experimentation with Hyper-V while ContainerD support is in alpha/beta stages.
+
+1. Add `node.kubernetes.io/windows-build` label using Windows kubelet
+2. Add RuntimeHandler to the CRI pull API
```

Comment:
I'm concerned we need this for others as well. For example if we use the `ImageStatus` call to determine if an image is present it will also need the `RuntimeHandler` to know if the "specific" version of the image is present. Otherwise it could return the appropriate status for the wrong version.